src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -15,25 +15,31 @@ import org.broadinstitute.dsde.workbench.sam.service.UserService._
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Try
+import pdi.jwt.{JwtOptions, JwtSprayJson}
+import spray.json._
+import java.time.Instant
 
 trait StandardUserInfoDirectives extends UserInfoDirectives {
   implicit val executionContext: ExecutionContext
 
   def requireUserInfo: Directive1[UserInfo] =
     (
-      headerValueByName(accessTokenHeader) &
-        headerValueByName(googleSubjectIdHeader) &
-        headerValueByName(expiresInHeader) &
-        headerValueByName(emailHeader)
+      optionalHeaderValueByName(accessTokenHeader) &
+        optionalHeaderValueByName(googleSubjectIdHeader) &
+        optionalHeaderValueByName(expiresInHeader) &
+        optionalHeaderValueByName(emailHeader) &
+        headerValueByName(authorizationHeader)
     ) tflatMap {
-      case (token, googleSubjectId, expiresIn, email) =>
-        onSuccess(
-          Try(expiresIn.toLong).fold[Future[UserInfo]](
-            t =>
-              Future.failed(
-                new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"expiresIn $expiresIn can't be converted to Long because  of $t"))),
-            l => getUserInfo(OAuth2BearerToken(token), GoogleSubjectId(googleSubjectId), WorkbenchEmail(email), l, directoryDAO).unsafeToFuture()
-          ))
+      case (Some(token), Some(googleSubjectId), Some(expiresIn), Some(email), _) =>
```

Comment:
Is that possible we are receiving OidcHeaders comes with authorizationHeader?

Reply:
yes and it definitely will. The `_` means that we don't care what it is